### PR TITLE
Bureau and Post Filters Redesign & Post Location Restructure

### PR DESCRIPTION
**Pull request by mjoyce91**
Redesigns the the bureau and post filters based on new designs. Makes use of the new Functional Bureau mapping. Also updates the sitewide use of `post.location` from a string to an object.

Relies on https://github.com/18F/State-TalentMAP-API/pull/420, which is good to be merged as soon as this PR is merged.

### CODE REVIEW

File: src/Constants/BureauMapping.js [hunk 1]
```diff
@@ -0,0 +1,51 @@
+export const BUREAU_MAPPING = [
+  /* {
```
[the remark this answers is not in the document]
Comment (mjoyce91) [reply src/Constants/BureauMapping.js]: Oh, I actually need to delete this entire file. We decided to handle all of this on the API.

File: src/Components/BidTracker/OverlayAlert/DraftAlert/DraftAlert.jsx [hunk 2]
```diff
@@ -24,9 +25,9 @@ class DraftAlert extends Component {
     const position = bid.position;
     const positionTitle = position.title;
     const post = position && position.post && position.post.location ?
-      position.post.location : NO_POST;
-    const skillCode = position && position.post ? position.post.location : NO_SKILL;
-    const grade = position && position.grade ? position.grade : NO_GRADE;
+      getPostName(position.post.location) : NO_POST;
+    const skillCode = position.skill ? position.skill : NO_SKILL;
```
Comment (burgwyn) [inline src/Components/BidTracker/OverlayAlert/DraftAlert/DraftAlert.jsx]: Good bug fix

File: src/Components/BidTracker/OverlayAlert/DraftAlert/DraftAlert.jsx [hunk 3]
```diff
@@ -24,9 +25,9 @@ class DraftAlert extends Component {
     const position = bid.position;
     const positionTitle = position.title;
     const post = position && position.post && position.post.location ?
-      position.post.location : NO_POST;
-    const skillCode = position && position.post ? position.post.location : NO_SKILL;
-    const grade = position && position.grade ? position.grade : NO_GRADE;
+      getPostName(position.post.location) : NO_POST;
```
Comment (burgwyn) [inline src/Components/BidTracker/OverlayAlert/DraftAlert/DraftAlert.jsx]: `getPostName(position.post)`

File: src/Components/BidTracker/OverlayAlert/DraftAlert/__snapshots__/DraftAlert.test.jsx.snap [hunk 4]
```diff
@@ -49,15 +49,15 @@ exports[`DraftAlertComponent matches snapshot 1`] = `
         >
           Post: 
         </span>
-        Curacao
+        00A
```
Comment (burgwyn) [inline src/Components/BidTracker/OverlayAlert/DraftAlert/__snapshots__/DraftAlert.test.jsx.snap]: This is be a City, Country string, right?

Comment (mjoyce91) [reply src/Components/BidTracker/OverlayAlert/DraftAlert/__snapshots__/DraftAlert.test.jsx.snap]: Yes, based on making the change you pointed out from changing `getPostName(position.post.location)` to `getPostName(position.post)`, this was fixed. I use `code` as a fallback.

File: src/Components/BidTracker/OverlayAlert/DraftAlert/DraftAlert.jsx (hunk 3 above)
Comment (mjoyce91) [reply src/Components/BidTracker/OverlayAlert/DraftAlert/DraftAlert.jsx]: Fixed!

**Pull request merged**